http-filter-example/header_processor.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+#include <string>
+#include <vector>
+
+namespace Envoy {
+namespace Http {
+
+class HeaderProcessor {
+public:
+  virtual ~HeaderProcessor() {};
+  virtual int parseOperation(std::vector<absl::string_view>& operation_expression) = 0;
+  virtual int executeOperation(RequestHeaderMap& headers) const = 0;
+  virtual int evaluateCondition() = 0;
+  bool getCondition() const { return condition_; }
+  void setCondition(bool result) { condition_ = result; }
+
+protected:
+  bool condition_;
+};
+
+class SetHeaderProcessor : public HeaderProcessor {
+public:
+  SetHeaderProcessor();
+  virtual ~SetHeaderProcessor() {}
+  virtual int parseOperation(std::vector<absl::string_view>& operation_expression);
+  virtual int executeOperation(RequestHeaderMap& headers) const;
+  virtual int evaluateCondition(); // TODO: will need to pass http-related metadata in order to evaluate dynamic values
+
+  const std::string& getKey() const { return header_key_; }
+  const std::vector<std::string>& getVals() const { return header_vals_; }
```

Comment:
Mind adding comments that the values returned by these functions must not outlive the `SetHeaderProcessor` object?